lib/puppetserver/ca/config/puppet.rb

```diff
@@ -152,12 +152,20 @@ def resolve_settings(overrides = {})
             settings[setting_name] = setting_value
           end
 
+          # If user provides subject-alt-names, we need to add the certname in addition
+          # to these.
+          if overrides[:dns_alt_names]
+            overrides[:dns_alt_names] = overrides[:dns_alt_names] + ',' + '$certname'
+          end
           # rename dns_alt_names to subject_alt_names now that we support IP alt names
-          settings[:subject_alt_names] = overrides.fetch(:dns_alt_names, "puppet,$certname")
+          settings[:subject_alt_names] = overrides.fetch(:dns_alt_names, "")
+          # If the cert is a master cert, it requires different defaults:
+          settings[:ca_subject_alt_names] = overrides.fetch(:dns_alt_names, "puppet,$certname")
```

Comment:
Instead of doing this here with a second setting, can we add the extra `$certname` and supply these defaults if necessary in the `local_ca` object when generating the master cert?

Reply:
Great idea.  Thanks!